Tasks/NuGetAuthenticateV1/main.ts

```diff
@@ -7,24 +7,52 @@ import { installCredProviderToUserProfile, configureCredProvider, configureCredP
 import { ProtocolType } from 'azure-pipelines-tasks-artifacts-common/protocols';
 import { getPackagingServiceConnections } from 'azure-pipelines-tasks-artifacts-common/serviceConnectionUtils'
 import { emitTelemetry } from 'azure-pipelines-tasks-artifacts-common/telemetry'
+import { ServiceConnection } from 'azure-pipelines-tasks-artifacts-common/serviceConnectionUtils';
 
 async function main(): Promise<void> {
     let forceReinstallCredentialProvider = null;
     let federatedFeedAuthSuccessCount: number = 0;
 
+    var feedUrl;
+    var entraWifServiceConnectionName;
+    var serviceConnections;
+
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+#if WIF
+
+        feedUrl = tl.getInput("feedUrl");
+        entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+        serviceConnections = getPackagingServiceConnections('nuGetServiceConnections');
+
+        // Failure case: User provides inputs for both NuGet & WIF Service Connections
+        if (serviceConnections.length > 0 && entraWifServiceConnectionName) {
+            tl.setResult(tl.TaskResult.Failed, tl.loc("Error_NuGetWithWIFNotSupported"));
+            return;
+        }
+
+        // Warning case: User provides feedUrl without providing a WIF service connection 
+        // In the future, we will shift to breaking behavior
+        if (entraWifServiceConnectionName) {
+            // Happy path, continue with flow
+        } else if (feedUrl) {
+            tl.warning(tl.loc("Warn_IgnoringFeedUrl"));
+            feedUrl = null;
+            // tl.setResult(tl.TaskResult.SucceededWithIssues, tl.loc("Error_NuGetWithFeedUrlNotSupported"));
+        }
+
+        // Validate input is valid feed URL
+        if (feedUrl && !validateFeedUrl(feedUrl)) {
+            tl.setResult(tl.TaskResult.Failed, tl.loc("Error_InvalidFeedUrl", feedUrl));
+        }
+
         // Install the credential provider
         forceReinstallCredentialProvider = tl.getBoolInput("forceReinstallCredentialProvider", false);
```

Comment:
This force reinstall credential provider should be outside ( I advise before) the #WIF block to ensure we are installing across all versions

Reply:
Oversight on my part. This has been moved back to its original spot before the WIF check